Mathlib/Analysis/Normed/Module/Completion.lean

```diff
@@ -100,6 +101,24 @@ instance [SeminormedCommRing A] [NormedAlgebra 𝕜 A] [UniformContinuousConstSM
         simp only [← coe_smul, norm_coe]
         exact norm_smul_le r x }
 
+instance [NormedField A] [CompletableTopField A] :
+    NormedField (UniformSpace.Completion A) where
+  toField := UniformSpace.Completion.instField
+  dist_eq x y := by
+    refine Completion.induction_on₂ x y ?_ ?_ <;> clear x y
+    · exact isClosed_eq (Completion.uniformContinuous_extension₂ _).continuous
+        (Continuous.comp Completion.continuous_extension continuous_sub)
+    · intro x y
+      rw [← Completion.coe_sub, norm_coe, Completion.dist_eq, dist_eq_norm]
+  norm_mul' x y := by
+    refine Completion.induction_on₂ x y ?_ ?_ <;> clear x y
+    · exact isClosed_eq (Continuous.comp continuous_norm continuous_mul)
+        (Continuous.comp _root_.continuous_mul
+          (Continuous.prod_map continuous_norm continuous_norm))
+    · intro x y
+      simp only [← coe_mul, norm_coe]
+      exact norm_mul x y
+
```

Comment:
Notes:
1. You don't need to include `toField` because that's already an instance
2. `fun_prop` can solve a lot of continuity-related goals here, as evidenced below.
3. a judicious `simp` call can clear those secondary goals more easily.
4. `clear` is not often something you actually need.
5. Since the `UniformSpace.Completion` namespace is open, you don't need to prefix `Completion` everywhere

```suggestion
  dist_eq x y := by
    refine induction_on₂ x y ?_ (by simp [← coe_sub, dist_eq_norm])
    exact isClosed_eq (uniformContinuous_extension₂ _).continuous (by fun_prop)
  norm_mul' x y := induction_on₂ x y (isClosed_eq (by fun_prop) (by fun_prop)) (by simp [← coe_mul])
```

Reply:
Thanks! changes made